### PR TITLE
Expose all 'feature flagged' kimchi gates to pickles

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR uses the `EnabledIf` constructor introduced in https://github.com/o1-labs/proof-systems/pull/889 to expose all of the new kimchi gates to pickles.

This dramatically reduces the amount of manual copying and updating needed to update pickles for a new gate.

Still TODO (on the kimchi side):
* add support for lookups in the `EnabledIf` logic, switch the old finalization to use it
* hook in the new optional features (currently hard-coded off)

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles/plonk_checks/scalars.ml [hunk 1]
```diff
@@ -1020,6 +1040,1680 @@ module Tick : S = struct
                        "0x40000000000000000000000000000000224698FC094CF91B992D30ECFFFFFFFB"
                    )
                  * cell (var (Witness 13, Curr)) ) ) )
+      ; ( Index ChaCha0
```
Comment (mrmr1993) [inline src/lib/pickles/plonk_checks/scalars.ml:1043]: This is autogenerated from the rust side!! I strongly recommend skipping during review!

File: src/lib/pickles/plonk_checks/gen_scalars/gen_scalars.ml [hunk 2]
```diff
@@ -84,6 +98,8 @@ module Env = struct
     ; beta : 'a
     ; gamma : 'a
     ; unnormalized_lagrange_basis : int -> 'a
+    ; enabled_if : Kimchi_types.feature_flag * (unit -> 'a) -> 'a
```
[the remark this answers is not in the document]
Comment (mimoo) [reply src/lib/pickles/plonk_checks/gen_scalars/gen_scalars.ml:101]: ah I guess you have a thunk, that you only evaluate if some feature flag is enabled?

[the remark this answers is not in the document]
Comment (mimoo) [reply src/lib/pickles/plonk_checks/gen_scalars/gen_scalars.ml:101]: (but in this case it should return an `'a option`)

[the remark this answers is not in the document]
Comment (mrmr1993) [reply src/lib/pickles/plonk_checks/gen_scalars/gen_scalars.ml:101]: Yea!

**Review by mimoo — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**